File: futa_be/server/src/models/danhSachBenXe.model.js
// src/models/user.model.js
import mongoose from 'mongoose';

const danhSachBenXe = new mongoose.Schema({
  name: { type: String, required: true },
  lat: { type: Number, required: true, unique: true },
  lng: { type: Number, required: true },
  province_id: { type: Number, required: true },
});

export default mongoose.model('danhSachBenXe', danhSachBenXe,'benXe');

File: futa/lib/features/datVeXe/data/models/danhSachBenXe_Model.dart

import '../../domain/entities/danhSachBenXe_ETT.dart';

class danhSachBenXeModel extends danhSachBenXe {


  danhSachBenXeModel({
    required String name,
    required double lat,
    required double lng,
    required int province_id,
  }) : super(name: name, lat: lat, lng: lng, province_id: province_id);

  factory danhSachBenXeModel.fromJson(Map<String, dynamic> json) {
    return danhSachBenXeModel(
      name: json['name'],
      lat: json['lat'],
      lng: json['lng'],
      province_id: json['province_id'],
    );
  }

  Map<String, dynamic> toJson() => {
    'name': name,
    'lat': lat,
    'lng': lng,
    'province_id': province_id
  };
}
